src/init/starship.zsh

```diff
@@ -94,4 +94,5 @@ setopt promptsubst
 
 PROMPT='$(::STARSHIP:: prompt --terminal-width="$COLUMNS" --keymap="${KEYMAP:-}" --status="$STARSHIP_CMD_STATUS" --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --cmd-duration="${STARSHIP_DURATION:-}" --jobs="$STARSHIP_JOBS_COUNT")'
 RPROMPT='$(::STARSHIP:: prompt --right --terminal-width="$COLUMNS" --keymap="${KEYMAP:-}" --status="$STARSHIP_CMD_STATUS" --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --cmd-duration="${STARSHIP_DURATION:-}" --jobs="$STARSHIP_JOBS_COUNT")'
+PROMPT2='$(::STARSHIP:: prompt --continuation --terminal-width="$COLUMNS" --keymap="${KEYMAP:-}" --status="$STARSHIP_CMD_STATUS" --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --cmd-duration="${STARSHIP_DURATION:-}" --jobs="$STARSHIP_JOBS_COUNT")'
```

Comment:
Unfortunately, this still causes an additional starship execution per prompt draw.

Reply:
Also, we *really* shouldn't need these other arguments right? This is just supposed to set a PS2 exactly once, when the shell launches?

Reply:
I think just removing the single quotes should be enough to get it to run once. And yeah since the other shells seem to use only `--continuation` without the rest of the flags.

```suggestion
PROMPT2=$(::STARSHIP:: prompt --continuation)
```

Reply:
It's been a long time since I messed around with the prompt variable quoting directly, but I seem to vaguely recall that double-quoting will work as well (which will be needed if e.g. someone tries to set `> >` as their continuation for some reason).

I don't have the slightest clue why that happens though, and someone should *definitely* double check me on that.

Reply:
Yes it should work as well. The behavior of shells is a bit weird in that area, and I'm not sure what POSIX mandates, but both Bash and Zsh won't do word splitting on variable assignment, and Zsh in particular doesn't do word splitting on any parameter/variable expansion.

Reply:
https://unix.stackexchange.com/questions/178294/is-field-splitting-and-filename-expansion-turn-off-in-rhs-of-variable-assignment

Though there are outlier shell in this regard due to how confusing this is.

Reply:
@chipbuster @segevfiner Thanks for testing this, I did not know that the prompt was still being updated after every command.